R/estimate_cox.R

```diff
@@ -0,0 +1,181 @@
+#' @param ... additional arguments passed on to the ellipsis of the call \code{survival::coxph(data = data, formula = survival::Surv(AVAL, 1-CNSR) ~ equation), ...)} .
+#'    Use \code{?survival::coxph} and \code{?survival::summary.coxph} for more information.
+#'
+#' @return a summarized coxph model
+#'
+#' @references \url{https://github.com/therneau/survival}
+#'
+#' @export
+#'
+#' @examples
+#'
+#' ## Null model
+#' visR::estimate_cox(data = adtte)
+#'
+#' ## Univariate Cox regression model using the covariate: `TRTP`
+#' # Stratified Kaplan-Meier analysis by `TRTP`
+#' visR::estimate_cox(data = adtte, equation = "TRTP")
+#'
+#' ## Multivariate Cox regression model using the covariates: `TRTP` and `SEX`
+#' visR::estimate_cox(data = adtte, equation = c("TRTP", "SEX"))
+#'
+#' ## Stratified Cox model (strata with one level)
+#' visR::estimate_cox(data = adtte, equation = c("AGE", "survival::strata(RACE)"))
+#'
+#' ## Null model on subset of adtte
+#' visR::estimate_cox(data = adtte[adtte$SEX == "F", ])
+#'
+#' ## Modify the default analysis by using the ellipsis
+#' visR::estimate_cox(data = adtte, equation = c( "SEX", "survival::cluster(USUBJID)"),
+#'   tier = "breslow", conf.int = 0.9)
+#'
+#' ## Example working with non CDISC data
+#' head(survival::veteran)
+#'
+#' # convert time and censoring data to ADaM variables
+#' # convert censoring status to CDISC principles
+#' veteran_adam <- survival::veteran %>%
+#'  dplyr::mutate(AVAL = time,
+#'                CNSR = dplyr::if_else(status == 1, 0, 1)
+#'  )
+#'
+#' visR::estimate_cox(data = veteran_adam, equation = c("trt", "diagtime"))
+
+
+
+estimate_cox <- function(
+  data = NULL
+  ,equation = NULL
+  ,CNSR = "CNSR"
+  ,AVAL = "AVAL"
+  ,conf.int = 0.95
+  ,...
+){
+
+  # Capture input to validate user input for data argument -----------------
+
+  ## Get actual data name as symbol
+  ### Magrittre pipe returns "." which inactivates recalls to survfit in downstream functions
+  ### map passes .x as cox_ccall$data
+  ### df: catch expressions that represent base R subsets
+  cox_call <- as.list(match.call())
+  dots <- list(...)
+  df_expr <- cox_call[["data"]]
+
+  ## Validate `data` and capture data name
+
+  if (is.null(data)) stop(paste0("data can't be NULL."))
+
+  if (base::length(base::deparse(cox_call[["data"]])) == 1 && base::deparse(cox_call[["data"]]) %in% c(".", ".x")) {
+    df <- the_lhs()
+    cox_call[["data"]] <- as.symbol(df)
+  } else {
+    df <- as.character(sub("\\[.*$", "", deparse(df_expr))[1])
+  }
+
+  if (!(inherits(data, "data.frame") | inherits(data, "tibble") | inherits(data, "data.table"))) {
+    stop(paste0("data can be of class `data.frame` or `tibble` or `data.table`."))
+  }
+
+  data <- as.data.frame(data)
+
+  # Validate columns --------------------------------------------------------
+
+  reqcols <- c(equation, CNSR, AVAL)
+  ss <- c("strata", "tt", "frailty", "ridge", "pspline", "cluster", "I", "factor")
```

Comment:
I dont like following code because the function will evolve and some arguments in coxph will get depricated.

I would replace this part with something more defensive or don't overengineer it:
available <- names(as.list(args(survival::coxph)))

requested <- names(as.list(args(survival::survfit)))

intersect(requested, available)